component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/SpaceDAOImpl.java

```diff
@@ -88,4 +88,46 @@ public SpaceEntity getSpaceByPrettyName(String spacePrettyName) {
     }
   }
 
+  @Override
+  public List<SpaceEntity> getCommonSpaces(String userId, String otherUserId, int offset, int limit) {
+    if (userId == null || userId == "") {
+      throw new IllegalArgumentException("the userId is null or equals to 0");
+    }
+    if (otherUserId == null || otherUserId == "") {
+      throw new IllegalArgumentException("the otherUserId is null or equals to 0");
+    }
+    if (offset < 0) {
+      throw new IllegalArgumentException("offset must be positive");
+    }
+    if (limit <= 0) {
+      throw new IllegalArgumentException("limit must be > 0");
+    }
+    TypedQuery<SpaceEntity> query = getEntityManager().createNamedQuery("SpaceEntity.getCommonSpacesBetweenTwoUsers",
+            SpaceEntity.class);
+    query.setParameter("userId", userId);
+    query.setParameter("otherUserId", otherUserId);
+    query.setFirstResult(offset);
+    query.setMaxResults(limit);
+    return query.getResultList();
+
+  }
+
+  @Override
+  public int countCommonSpaces(String userId, String otherUserId) {
+    if (userId == null || userId == "") {
+      throw new IllegalArgumentException("userId is null or equals to 0");
+    }
+    if (otherUserId == null || otherUserId == "") {
+      throw new IllegalArgumentException("otherUserId is null or equals to 0");
+    }
+
+    Query q = getEntityManager().createNativeQuery("SELECT count(*) FROM SOC_SPACES t where t.SPACE_ID in " +
+            "( SELECT distinct t1.SPACE_ID FROM SOC_SPACES_MEMBERS t1  " +
+            "where t1.USER_ID = :userId  " +
+            "and (SELECT distinct t2.SPACE_ID FROM SOC_SPACES_MEMBERS t2 " +
+            "where t1.SPACE_ID=t2.SPACE_ID and t2.USER_ID = :otherUserId ))");
+    q.setParameter("userId", userId);
+    q.setParameter("otherUserId", otherUserId);
+    return ((Number) q.getSingleResult()).intValue();
```

Comment:
If you use a named  query, and typing it with a long (like here : https://github.com/Meeds-io/social/blob/b5f2348ecb1ce304d605423520c139b213b28587/component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java#L188-L195) the cast is not necessary